File: cartLoc.py
from playwright.sync_api import Page


class cartLoc:
    """Locators for Product and Cart Pages"""
    
    def __init__(self, page: Page):
        self.page = page
        
        # Product page elements
        self.product_title = page.locator("h1")
        self.product_price = page.locator(".card-title").filter(has_text="$")
        self.quantity_input = page.locator("[data-test='quantity']")
        self.increase_quantity_btn = page.locator("[data-test='increase-quantity']")
        self.decrease_quantity_btn = page.locator("[data-test='decrease-quantity']")
        self.add_to_cart_btn = page.locator("[data-test='add-to-cart']")
        
        # Cart page elements
        self.nav_cart = page.locator("[data-test='nav-cart']")
        self.cart_badge = page.locator(".badge")
        self.cart_product_title = page.locator("[data-test='product-title']")
        self.cart_product_quantity = page.locator("[data-test='product-quantity']")
        self.cart_product_price = page.locator("[data-test='product-price']")
        self.cart_total = page.locator("[data-test='cart-total']")
        self.proceed_btn = page.locator("[data-test='proceed-1']")
        self.continue_shopping_btn = page.locator("[data-test='continue-shopping']")

File: Locators/cartLoc.js
const { expect } = require('@playwright/test');

class CartLocators {
    /**
     * Locators for Product and Cart Pages
     * @param {import('@playwright/test').Page} page
     */
    constructor(page) {
        this.page = page;

        // Product page elements
        this.productTitle = page.locator('h1');
        this.productPrice = page.locator('.card-title').filter({ hasText: '$' });
        this.quantityInput = page.locator("[data-test='quantity']");
        this.increaseQuantityBtn = page.locator("[data-test='increase-quantity']");
        this.decreaseQuantityBtn = page.locator("[data-test='decrease-quantity']");
        this.addToCartBtn = page.locator("[data-test='add-to-cart']");

        // Cart page elements
        this.navCart = page.locator("[data-test='nav-cart']");
        this.cartBadge = page.locator("[data-test='nav-cart'] .badge");
        this.cartProductTitle = page.locator("[data-test='product-title']");
        this.cartProductQuantity = page.locator("[data-test='product-quantity']");
        this.cartProductPrice = page.locator("[data-test='product-price']");
        this.cartTotal = page.locator("[data-test='cart-total']");
        this.proceedBtn = page.locator("[data-test='proceed-1']");
        this.continueShoppingBtn = page.locator("[data-test='continue-shopping']");
    }
}

module.exports = { CartLocators };
